### ISSUE TITLE
webpack cannot statically wire moment

### ISSUE THREAD

**Issue by vlahupetar** [MadMG/moment-jdateformatparser#11]
Hi,
When using moment-jdateformatparser with webpack, webpack throws an error in the section where moment-jdateformatparser is trying to determine whether the environment is node or AMD. 

not sure about a solution but i thought i would send you guys a heads up about it

here is the warning as it comes from webpack
WARNING in ./~/moment-jdateformatparser/moment-jdateformatparser.js
Critical dependencies:
163:78-85 require function is used in a way in which dependencies cannot be statically extracted
 @ ./~/moment-jdateformatparser/moment-jdateformatparser.js 163:78-85

**Comment by herom** [comment]
Thanks for reporting this issue @vlahupetar - I'll keep this on my radar and see what I can do to make the integration with webpack work :+1:

**Comment by VladimirKorablev** [comment]
Hi! I had same issue with webpack.
I've fixed it changing moment detection 
FROM:
```
if (typeof this.moment === 'undefined' && typeof require !== 'undefined' && require !== null) {
  if (typeof module !== 'undefined' && typeof module.exports !== 'undefined') { //Check if the environment is Node.js
    hookMoment(require('moment')); //if it is, we have to require it different (without the surrounding Array)
  } else {
    require(['moment'], function (moment) {
      hookMoment(moment);
    });
  }
} else {
  hookMoment(this.moment);
}
```

TO (like in moment-duration-format module):
```
if (typeof require === "function") {
	try { moment = require('moment'); }
	catch (e) {}
}

if (!moment && root.moment) {
	moment = root.moment;
}

if (!moment) {
	throw "Moment JDateFormatParser cannot find Moment.js";
}
hookMoment(moment);
```

**Comment by herom** [comment]
@VladimirKorablev would you be able to make a PR to implement your solution? :+1:

**Comment by getfatday** [comment]
@herom Updated the UMD loader, so it matches the general specification at [umdjs/umd](https://github.com/umdjs/umd). This also resolves the webpack issues.

**Comment by amannn** [comment]
I've worked around the issue with this:

```js
window.moment = moment;
require('imports?module=>false&require=>false&define=>false&this=>window!moment-jdateformatparser');
```

Hopefully that PR gets merged soon.

**Comment by Southpaw17** [comment]
Is there any update on this?

**Comment by herom** [comment]
Closed due to the merge of #15
